contrib/react-native-debugging.md

```diff
@@ -0,0 +1,110 @@
+
+# React Native Debugging
+
+## Preface
+
+When developing Realm, there are times where the node tests and debugging methods are not enough.  If you have tests failing in React Native, you will need a way to set breakpoints in both the JS and C++ code.  This document will explain how to make that happen.  Note: this is currently only testing iOS in the simulator.
+
+## Preparation and Assumptions
+It is assumed the developer has their machine prepared to build realm and react-native projects.  Refer to [building.md](./building.md) if that is not the case.
+
+In your realm-js project, run the following commands to prepare for debug mode:
+```
+# install js dependencies, but skip building the node binaries
+$ npm install --ignore-scripts
+
+# build the xcframework
+$ ./scripts/build-ios.sh -c Debug -s
+```
+
+You are now prepared to either use the [ReactTestApp](#setup-reacttestapp) or [prepare a custom project](#setup-custom-react-native-project)
+
+## Setup ReactTestApp
+
+There is a test application in `tests/ReactTestApp` which will run all the unit tests living in `tests/js`.  This is a good place to start.  It is already setup to debug the js code and the c++ source is included by reference in the Xcode project.  However this will require some steps before it is ready to run.
```

Comment:
```suggestion
There is a test application in `tests/ReactTestApp` which will run all the unit tests living in `tests/js`.  This is a good place to start.  It is already setup to debug the JS code and the C++ source is included by reference in the Xcode project.  However this will require some steps before it is ready to run.
```